clusters/resource_cluster.go

```diff
@@ -241,14 +253,46 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 	common.DataToStructPointer(d, clusterSchema, &cluster)
 	var clusterInfo ClusterInfo
 	var err error
+
 	if hasClusterConfigChanged(d) {
 		log.Printf("[DEBUG] Cluster state has changed!")
 		if err := cluster.Validate(); err != nil {
 			return err
 		}
 		cluster.ModifyRequestOnInstancePool()
 		fixInstancePoolChangeIfAny(d, &cluster)
-		clusterInfo, err = clusters.Edit(cluster)
+
+		// We can only call the resize api if the cluster is in the running state
+		// and only the cluster size (ie num_workers OR autoscale) is being changed
+		hasNumWorkersChanged := d.HasChange("num_workers")
+		hasAutoscaleChanged := d.HasChange("autoscale")
+		hasOnlyResizeClusterConfigChanged := hasOnlyResizeClusterConfigChanged(d)
+
+		isNumWorkersResizeForNonAutoscalingCluster := hasOnlyResizeClusterConfigChanged && hasNumWorkersChanged && !hasAutoscaleChanged
+		isAutoscaleConfigResizeForAutoscalingCluster := hasOnlyResizeClusterConfigChanged && hasAutoscaleChanged && !hasNumWorkersChanged
+		isNonAutoScalingToAutoscalingResize := hasOnlyResizeClusterConfigChanged && hasAutoscaleChanged && hasNumWorkersChanged && cluster.Autoscale != nil
+		isAutoScalingToNonAutoscalingResize := hasOnlyResizeClusterConfigChanged && hasAutoscaleChanged && hasNumWorkersChanged && cluster.Autoscale == nil
+		
+		isValidClusterConfigChangeForResizeAPI := isNumWorkersResizeForNonAutoscalingCluster || isAutoscaleConfigResizeForAutoscalingCluster || isNonAutoScalingToAutoscalingResize || isAutoScalingToNonAutoscalingResize
```

Comment:
i think that we can simplify the logic by removing this variable.

Reply:
Yeah, nice catch, logic much cleaner now